### ISSUE TITLE
release scalacheck_sjs0.6_2.13.0-M1

### ISSUE THREAD

**Issue by xuwei-k** [rickynils/scalacheck#330]
@rickynils

**Comment by xuwei-k** [comment]
ping

**Comment by djspiewak** [comment]
This is actually holding up a good chunk of the ecosystem, since most projects cannot release their SJS 2.13.0-M1 cross-builds without ScalaCheck and ScalaTest.

**Comment by rickynils** [comment]
The `doc` task seems to fail due to changes in ScalaJS (bumped to 0.6.17 to get the 2.13.0-M1 release):

```
> doc
[info] Main Scala API documentation to /home/rickard/src/scalacheck/js/target/scala-2.13.0-M1/api...
[error] /home/rickard/src/scalacheck/src/main/scala/org/scalacheck/Prop.scala:25: @JSExportDescendentClasses is deprecated and will be removed in 1.0.0. For use cases where you want to simulate "reflective" instantiation, use @EnableReflectiveInstantion and scala.scalajs.reflect.Reflect.lookupInstantiatableClass instead.
[error]   (you can suppress this warning in 0.6.x by passing the option `-P:scalajs:suppressExportDeprecations` to scalac)
[error] @Platform.JSExportDescendentClasses
[error]  ^
[error] /home/rickard/src/scalacheck/src/main/scala/org/scalacheck/Prop.scala:26: @JSExportDescendentObjects is deprecated and will be removed in 1.0.0. For use cases where you want to simulate "reflective" loading, use @EnableReflectiveInstantion and scala.scalajs.reflect.Reflect.lookupLoadableModuleClass instead.
[error]   (you can suppress this warning in 0.6.x by passing the option `-P:scalajs:suppressExportDeprecations` to scalac)
[error] @Platform.JSExportDescendentObjects
[error]  ^
[error] /home/rickard/src/scalacheck/src/main/scala/org/scalacheck/Properties.scala:29: @JSExportDescendentClasses is deprecated and will be removed in 1.0.0. For use cases where you want to simulate "reflective" instantiation, use @EnableReflectiveInstantion and scala.scalajs.reflect.Reflect.lookupInstantiatableClass instead.
[error]   (you can suppress this warning in 0.6.x by passing the option `-P:scalajs:suppressExportDeprecations` to scalac)
[error] @Platform.JSExportDescendentClasses
[error]  ^
[error] /home/rickard/src/scalacheck/src/main/scala/org/scalacheck/Properties.scala:30: @JSExportDescendentObjects is deprecated and will be removed in 1.0.0. For use cases where you want to simulate "reflective" loading, use @EnableReflectiveInstantion and scala.scalajs.reflect.Reflect.lookupLoadableModuleClass instead.
[error]   (you can suppress this warning in 0.6.x by passing the option `-P:scalajs:suppressExportDeprecations` to scalac)
[error] @Platform.JSExportDescendentObjects
[error]  ^
[info] No documentation generated with unsuccessful compiler run
[error] four errors found
[error] (js/compile:doc) Scaladoc generation failed
[error] Total time: 6 s, completed Jun 8, 2017 4:37:47 PM
```

**Comment by djspiewak** [comment]
Ping @sjrd.

**Comment by xuwei-k** [comment]
just remove `-Xfatal-warnings` ?

https://github.com/rickynils/scalacheck/blob/1.13.5/build.sbt#L51

**Comment by djspiewak** [comment]
That's an option: https://github.com/djspiewak/scalacheck/tree/wip/2.13  It'll have to be addressed eventually though.

**Comment by SethTisue** [comment]
http://www.scala-js.org/news/2017/03/21/announcing-scalajs-0.6.15/ explains the deprecation and how to migrate

**Comment by sjrd** [comment]
As pointed by @SethTisue, the 0.6.15 announcement should explain everything you need to know to migrate. Feel free to ask further questions if it's not enough.

**Comment by SethTisue** [comment]
@rickynils is your wip on this on GitHub? I could try to finish it up.

**Comment by rickynils** [comment]
@SethTisue No wip. I just bumbed the scalajs version in plugins.sbt. If PRs are provided for the 1.13.5_sonatype and 1.12.6_sonatype branches I can try to publish releases. But maybe we have to cut new releases if there are code changes.

Sorry for my unresponsiveness. It is mainly because I got a kid some weeks ago, and partly because I have never used ScalaJS.

**Comment by SethTisue** [comment]
please continue to prioritize your human children over your digital ones :-)

the release notes say that `scalacOptions += "-P:scalajs:suppressExportDeprecations"` will continue to work for Scala.js 0.6.x, so that seems like an acceptable minimal fix for now. I will send PRs on the two branches you suggest

**Comment by SethTisue** [comment]
trio of pull requests: #339, #340, #341

**Comment by SethTisue** [comment]
hmm, actually it's not that hard to go ahead and make the necessary changes to not use the deprecated stuff at all. revising the PRs now

**Comment by SethTisue** [comment]
hmm, well, probably better to make the minimal change on the release branches. I did the better version on master (#341); the other two are fine as-is I think.

**Comment by xuwei-k** [comment]
Thanks!
https://oss.sonatype.org/content/repositories/releases/org/scalacheck/scalacheck_sjs0.6_2.13.0-M1/

**Comment by rickynils** [comment]
@SethTisue Thanks for the PRs! I've published releases for ScalaCheck 1.12.6 and 1.13.5
